Spigot-Server-Patches/0619-added-PlayerTradeEvent.patch

```diff
@@ -4,72 +4,127 @@ Date: Thu, 2 Jul 2020 16:12:10 -0700
 Subject: [PATCH] added PlayerTradeEvent
 
 
-diff --git a/src/main/java/net/minecraft/world/entity/EntityInsentient.java b/src/main/java/net/minecraft/world/entity/EntityInsentient.java
-index 89d24d7532a256434513a45c901946e28db396bd..ff482e3774f580d8ba7028f6c5141888d3bd907a 100644
---- a/src/main/java/net/minecraft/world/entity/EntityInsentient.java
-+++ b/src/main/java/net/minecraft/world/entity/EntityInsentient.java
-@@ -93,7 +93,7 @@ import org.bukkit.event.entity.EntityUnleashEvent.UnleashReason;
- public abstract class EntityInsentient extends EntityLiving {
- 
-     private static final DataWatcherObject<Byte> b = DataWatcher.a(EntityInsentient.class, DataWatcherRegistry.a);
--    public int e;
-+    public int e;public void setAmbientSoundTime(int time) { this.e = time; } // Paper - OBFHELPER
-     protected int f;
-     protected ControllerLook lookController;
-     protected ControllerMove moveController;
-@@ -295,6 +295,7 @@ public abstract class EntityInsentient extends EntityLiving {
-         this.datawatcher.register(EntityInsentient.b, (byte) 0);
-     }
- 
-+    public int getAmbientSoundInterval() { return D(); } // Paper - OBFHELPER
-     public int D() {
-         return 80;
-     }
 diff --git a/src/main/java/net/minecraft/world/entity/npc/EntityVillagerAbstract.java b/src/main/java/net/minecraft/world/entity/npc/EntityVillagerAbstract.java
-index 10067731e6289211cb11b2b368527a1c45603314..49821f0f26864c35be84a4a4288857a04668fbba 100644
+index 10067731e6289211cb11b2b368527a1c45603314..5694e60b14db578c01334008d03e48be1191ed9f 100644
 --- a/src/main/java/net/minecraft/world/entity/npc/EntityVillagerAbstract.java
 +++ b/src/main/java/net/minecraft/world/entity/npc/EntityVillagerAbstract.java
-@@ -39,6 +39,9 @@ import org.bukkit.craftbukkit.inventory.CraftMerchantRecipe;
- import org.bukkit.entity.AbstractVillager;
- import org.bukkit.event.entity.VillagerAcquireTradeEvent;
- // CraftBukkit end
-+// Paper start
-+import io.papermc.paper.event.player.PlayerTradeEvent;
-+// Paper end
- 
- public abstract class EntityVillagerAbstract extends EntityAgeable implements NPC, IMerchant {
- 
-@@ -135,16 +138,27 @@ public abstract class EntityVillagerAbstract extends EntityAgeable implements NP
+@@ -133,17 +133,27 @@ public abstract class EntityVillagerAbstract extends EntityAgeable implements NP
+     @Override
+     public void setForcedExperience(int i) {}
  
++    // Paper start
++    @Override
++    public void processTrade(MerchantRecipe recipe, io.papermc.paper.event.player.PlayerTradeEvent event) { // The MerchantRecipe passed in here is the one set by the PlayerTradeEvent
++        if (event.willIncreaseTradeUses()) { recipe.increaseUses(); }
++        if (event.isRewardingExp()) { this.rewardTradeXp(recipe); }
```

Comment:
You don't need to keep it all on one line: do proper line breaks so this is readable